docs/content/concepts/partitions-schedules-sensors/sensors.mdx

```diff
@@ -205,6 +191,58 @@ def every_fifth_asset_c_sensor(context):
         return SkipReason(f"asset_c only materialized {len(asset_events)} times.")
 ```
 
+#### Triggering a partitioned run after upstream partitions are materialized
+
+The <PyObject object="MultiAssetSensorEvaluationContext" method="latest_materialization_records_by_partition"/> method fetches the latest materialization records by partition for a given asset. Using this method, you can trigger downstream partitioned runs when the corresponding partitions of upstream assets are materialized.
+
+The following example monitors two upstream daily-partitioned assets, kicking off materializations of the corresponding partition in the downstream daily-partitioned asset. Whenever a partition is replaced in `upstream_daily_1` and the same partition has previously been materialized in `upstream_daily_2` (and vice versa), the sensor yields a run request for the downstream asset.
+
+```python file=/concepts/partitions_schedules_sensors/sensors/sensors.py startafter=start_daily_partitioned_asset_with_two_upstreams endbefore=end_daily_partitioned_asset_with_two_upstreams
+@multi_asset_sensor(
+    asset_keys=[AssetKey("upstream_daily_1"), AssetKey("upstream_daily_2")],
+    job=downstream_daily_job,
+)
+def trigger_daily_asset_if_all_upstream_partitions_materialized(context):
+    run_requests = []
+
+    def evaluate_latest_partitions(asset_key):
+        # For each asset, get the latest materialization record for each partition.
+        other_asset_keys = [
+            other_key for other_key in context.asset_keys if other_key != asset_key
+        ]
+        for (
+            partition,
+            materialization,
+        ) in context.latest_materialization_records_by_partition(asset_key).items():
+            # Check that the partition is materialized in the other asset. If so,
+            # yield a run request for the downstream partition.
+            if all(
+                [
+                    context.all_partitions_materialized(other_key, [partition])
+                    for other_key in other_asset_keys
+                ]
+            ):
+                run_requests.append(
```

Comment:
I might be reading this incorrectly, but it seems like we might end up adding `RunRequest`s twice for the same partition?  E.g. if we depend on "asset1" and "asset2" and they both have a materialization for partition "2020-01-02", we'll call `evaluate_latest_partitions` on "asset1", which will add a `RunRequest` and then on "asset2", which will add another `RunRequest`?